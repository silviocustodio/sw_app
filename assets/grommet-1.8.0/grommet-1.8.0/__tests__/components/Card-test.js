// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Card from '../../src/js/components/Card';

// needed because this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('Card', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Card thumbnail="http://grommet.github.io/img/carousel-1.png"
        label="Featured Post"
        heading="The Key Steps to Reducing Software Spend" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with a non-default colorIndex', () => {
    const component = renderer.create(
      <Card colorIndex="light-2" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders a header with the set size', () => {
    const component = renderer.create(
      <Card label="Hello from Grommet" textSize="large" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when reverse', () => {
    const component = renderer.create(
      <Card label="Hello from Grommet" reverse={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when the direction is set to row', () => {
    const component = renderer.create(
      <Card label="Hello from Grommet" direction="row" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly when the headingStrong is false', () => {
    const component = renderer.create(
      <Card label="Hello from Grommet" headingStrong={false} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
