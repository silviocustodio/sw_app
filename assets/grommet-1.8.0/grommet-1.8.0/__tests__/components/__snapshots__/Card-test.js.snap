// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card has correct default options 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-medium grommetux-box--size grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--justify-center grommetux-box--align-center grommetux-box--responsive grommetux-box--basis-small grommetux-box--pad-none grommetux-box--flex-off grommetux-card__thumbnail"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={
      Object {
        "background": "url(http://grommet.github.io/img/carousel-1.png) no-repeat center center",
        "backgroundSize": "cover",
      }
    }
    tabIndex={undefined}
  />
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <label
      className="grommetux-label grommetux-label--uppercase grommetux-label--margin-none grommetux-label--medium"
      htmlFor={undefined}
    >
      Featured Post
    </label>
    <h2
      className="grommetux-heading grommetux-heading--strong"
    >
      The Key Steps to Reducing Software Spend
    </h2>
  </div>
</div>
`;

exports[`Card renders a header with the set size 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-medium grommetux-box--size grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <label
      className="grommetux-label grommetux-label--uppercase grommetux-label--margin-none grommetux-label--medium"
      htmlFor={undefined}
    >
      Hello from Grommet
    </label>
  </div>
</div>
`;

exports[`Card renders correctly when reverse 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--justify-between grommetux-box--reverse grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-medium grommetux-box--size grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <label
      className="grommetux-label grommetux-label--uppercase grommetux-label--margin-none grommetux-label--medium"
      htmlFor={undefined}
    >
      Hello from Grommet
    </label>
  </div>
</div>
`;

exports[`Card renders correctly when the direction is set to row 1`] = `
<div
  className="grommetux-box grommetux-box--direction-row grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-xlarge grommetux-box--size grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--basis-2-3 grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <label
      className="grommetux-label grommetux-label--uppercase grommetux-label--margin-none grommetux-label--medium"
      htmlFor={undefined}
    >
      Hello from Grommet
    </label>
  </div>
</div>
`;

exports[`Card renders correctly when the headingStrong is false 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-medium grommetux-box--size grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  >
    <label
      className="grommetux-label grommetux-label--uppercase grommetux-label--margin-none grommetux-label--medium"
      htmlFor={undefined}
    >
      Hello from Grommet
    </label>
  </div>
</div>
`;

exports[`Card renders correctly with a non-default colorIndex 1`] = `
<div
  className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-none grommetux-box--wrap grommetux-box--width-medium grommetux-box--size grommetux-background-color-index-light-2 grommetux-background-color-index--light grommetux-card"
  id={undefined}
  onClick={undefined}
  role={undefined}
  style={Object {}}
  tabIndex={undefined}
>
  <div
    className="grommetux-box grommetux-box--direction-column grommetux-box--responsive grommetux-box--pad-medium grommetux-card__content"
    id={undefined}
    onClick={undefined}
    role={undefined}
    style={Object {}}
    tabIndex={undefined}
  />
</div>
`;
